### PR TITLE
Feature add installation of golem in the virtual machine

### PR DESCRIPTION
**Pull request by bartoszbetka**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -361,3 +368,22 @@ concent-update.sh <branch>
...
 ```
 
+`golem-env.sh`
+- run virtualenv and change directory to `golem`
+
+```bash
+source golem-env.sh
+```
+
+`golem-run-console-mode.sh`
```
Comment (kbeker) [inline README.md]: Shouldn't you also run hyperg when you are trying to run golem in console mode?

File: concent-vm/vars.yml [hunk 2]
```diff
@@ -1,2 +1,5 @@
-concent_deployment_dir: /home/builder/concent-deployment
-concent_dir:            /home/vagrant/concent
+concent_deployment_dir:      /home/builder/concent-deployment
+concent_dir:                 /home/vagrant/concent
+golem_dir:                   /home/vagrant/golem
+golem_hyperdrive_repository: https://github.com/mfranciszkiewicz/golem-hyperdrive/releases/download/v0.2.6/hyperg_0.2.6_linux-x64.tar.gz
```
Comment (cameel) [inline concent-vm/vars.yml]: https://github.com/mfranciszkiewicz/golem-hyperdrive/releases/download/v0.2.6/hyperg_0.2.6_linux-x64.tar.gz is not a repository. It's a release package URL.

Please simply set `golem_hyperdrive_version` to `0.2.6` and put the rest of the path in the playbook. Or at least in a separate variable:

``` yam
hyperg_version: "0.2.6"
hyperg_url: https://github.com/mfranciszkiewicz/golem-hyperdrive/releases/download/v{{ hyperg_version }}/hyperg_{{ hyperg_version }}_linux-x64.tar.gz
```

File: concent-vm/vars.yml [hunk 3]
```diff
@@ -1,2 +1,5 @@
...
+golem_hyperdrive_checksum:   sha256:94220ef8533030652d4a1c126f06fa9cdce9d620b91f1c4574a109a3891e8adf
```
Comment (cameel) [inline concent-vm/vars.yml]: I'd skip the checksum. Updating it every time will be cumbersome. And this is just a machine for development so it's not like it matters much.

File: concent-vm/install-golem.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        - name:   Extract hyperg
+          unarchive:
+            src:        ~/hyperg.tar.gz
+            dest:       ~/
+            remote_src: yes
+
+        - name:    Copy to execute path
+          file:
+            src:  ~/hyperg/hyperg
+            dest: ~/bin/hyperg
```
Comment (cameel) [inline concent-vm/install-golem.yml]: Please reword: `Copy to execute path` -> `Install a symbolic link to hyperg in ~/bin/`

File: concent-vm/install-golem.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+          get_url:
+            url:      "{{ golem_hyperdrive_repository }}"
+            dest:     ~/hyperg.tar.gz
+            mode:     0644
+            checksum: "{{ golem_hyperdrive_checksum }}"
+
+        - name:   Extract hyperg
+          unarchive:
+            src:        ~/hyperg.tar.gz
+            dest:       ~/
```
Comment (cameel) [inline concent-vm/install-golem.yml]: Please remove the archive after extracting it.

File: concent-vm/install-golem.yml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+        - name:   Run setup.py develop
+          shell: ~/golem_virtualenv/bin/python setup.py develop
+          args:
+            chdir: "{{ golem_dir }}"
+
+        - name:   Download hyperg
+          get_url:
+            url:      "{{ golem_hyperdrive_repository }}"
+            dest:     ~/hyperg.tar.gz
```
Comment (cameel) [inline concent-vm/install-golem.yml]: I'd download it to `/tmp/`. We don't need it after the installation.

File: concent-vm/install-golem.yml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        - name:   Clone golem repository
+          git:
+            repo:    "{{ repositories.golem.url }}"
+            dest:    "{{ golem_dir }}"
+            version: "{{ golem_version }}"
+            clone:   yes
+            update:  yes
+
+        - name:   Create virtualenv for golem
+          command:  virtualenv ~/golem_virtualenv --python python3.6 creates="~/golem_virtualenv"
```
Comment (cameel) [inline concent-vm/install-golem.yml]: Ansible's [pip](https://docs.ansible.com/ansible/latest/modules/pip_module.html) command can create a virtualenv automatically if it does not exist.

File: concent-vm/install-golem.yml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+        - name:   Install requirements for golem
+          pip:
+            requirements:       "{{ golem_dir }}/{{ item }}"
+            virtualenv:         ~/golem_virtualenv
+            virtualenv_python:  python3.6
```
Comment (cameel) [inline concent-vm/install-golem.yml]: Use ``python_version`` variable.

File: concent-vm/install-golem.yml [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+            dest:    "{{ golem_dir }}"
+            version: "{{ golem_version }}"
+            clone:   yes
+            update:  yes
+
+        - name:   Create virtualenv for golem
+          command:  virtualenv ~/golem_virtualenv --python python3.6 creates="~/golem_virtualenv"
+
+        - name:   Install requirements for golem
+          pip:
```
Comment (cameel) [inline concent-vm/install-golem.yml]: It would be nice to say which particular requirements you're installing (golem has several different requirements files).

File: concent-vm/helper-scripts/golem-env.sh [hunk 10]
```diff
@@ -0,0 +1,5 @@
+#! /bin/bash -e
+
+source ~/golem_virtualenv/bin/activate
+cd ~/golem/
+
```
Comment (cameel) [inline concent-vm/helper-scripts/golem-env.sh]: Why the extra line?

File: concent-vm/configure.yml [hunk 11]
```diff
@@ -90,12 +104,6 @@
             state:  present
             groups: docker
 
-        - name:   Add docker group to vagrant user
-          user:
-            name:   vagrant
-            groups: docker
```
Comment (cameel) [inline concent-vm/configure.yml]: This should be in a separate commit.

Actually, why are you deleting it?

File: concent-vm/Vagrantfile [hunk 12]
```diff
@@ -62,6 +90,8 @@ Vagrant.configure("2") do |config|
         machineId = File.read(".vagrant/machines/default/virtualbox/id")
         run "VBoxManage storageattach '#{machineId}'" +
         " --storagectl 'SATA Controller' --port 1 --device 0 --type hdd --medium none"
+        run "VBoxManage storageattach '#{machineId}'" +
+        " --storagectl 'SATA Controller' --port 2 --device 0 --type hdd --medium none"
```
Comment (cameel) [inline concent-vm/Vagrantfile]: Please align this for readability.

File: concent-vm/Vagrantfile [hunk 13]
```diff
@@ -40,19 +46,41 @@ Vagrant.configure("2") do |config|
...
+        sudo usermod -aG docker vagrant
     SHELL
 
     config.trigger.after :halt do
         info "dettach drive"
         machineId = File.read(".vagrant/machines/default/virtualbox/id")
         run "VBoxManage storageattach '#{machineId}'" +
         " --storagectl 'SATA Controller' --port 1 --device 0 --type hdd --medium none"
+        run "VBoxManage storageattach '#{machineId}'" +
+        " --storagectl 'SATA Controller' --port 2 --device 0 --type hdd --medium none"
```
Comment (cameel) [inline concent-vm/Vagrantfile]: Please align this for readability.

File: concent-vm/Vagrantfile [hunk 14]
```diff
@@ -27,11 +27,17 @@ Vagrant.configure("2") do |config|
...
     blockchain_disk = './disk/blockchain_disk.vdi'
+    golem_blockchain_disk = './disk/golem_blockchain_disk.vdi'
+
     config.vm.provider "virtualbox" do |vb|
         if not File.exists?(blockchain_disk)
           vb.customize ['createhd', '--filename', blockchain_disk, '--variant', 'Fixed', '--size', 30 * 1024]
         end
+        if not File.exists?(golem_blockchain_disk)
+          vb.customize ['createhd', '--filename', golem_blockchain_disk, '--variant', 'Fixed', '--size', 1 * 1024]
+        end
```
Comment (cameel) [inline concent-vm/Vagrantfile]: Please fix indentation and align for redability.

File: concent-vm/Vagrantfile [hunk 15]
```diff
@@ -40,19 +46,41 @@ Vagrant.configure("2") do |config|
...
-        fi
-        mountpoint -q $directory || sudo mount /dev/sdb $directory
+        directories=(
+            /home/vagrant/blockchain
+            /home/vagrant/.local
+        )
+        partition_name=(
+            /dev/sdb
+            /dev/sdc
+        )
```
Comment (cameel) [inline concent-vm/Vagrantfile]: This is a list, name should be in plural (`partition_names`).

File: concent-vm/Vagrantfile [hunk 16]
```diff
@@ -40,19 +46,41 @@ Vagrant.configure("2") do |config|
...
+        for directory in "${directories[@]}"; do
+            if [ ! -d "$directory" ]; then
+                mkdir $directory
+            fi
+            mountpoint -q $directory || sudo mount ${partition_name[$index]} $directory
+            ((index++))
+        done
+        # Create docker group and add your user to it.
+        grep -q docker /etc/group || sudo groupadd docker
```
Comment (cameel) [inline concent-vm/Vagrantfile]: Please use long option names.

File: concent-vm/Vagrantfile [hunk 17]
```diff
@@ -40,19 +46,41 @@ Vagrant.configure("2") do |config|
...
+        sudo usermod -aG docker vagrant
```
Comment (cameel) [inline concent-vm/Vagrantfile]: Creating groups and users belongs in the playbook. Why do you need it here?
